config/src/main/java/com/typesafe/config/impl/SimpleConfig.java

```diff
@@ -295,7 +295,15 @@ public Long getBytes(String path) {
 
     @Override
     public ConfigMemorySize getMemorySize(String path) {
-        return ConfigMemorySize.ofBytes(getBytes(path));
+        BigInteger size;
+        try {
+            size = BigInteger.valueOf(getLong(path));
+        } catch (ConfigException.WrongType e) {
+            ConfigValue v = find(path, ConfigValueType.STRING);
+            size = parseBytesAsBigInteger((String) v.unwrapped(),
+                v.origin(), path);
+        }
+        return ConfigMemorySize.ofBytes(size);
```

Comment:
would be great to reduce cut-and-paste between getMemorySize and getBytes here ... maybe flip which calls the other, so make getBytes call getMemorySize.toBytes? 

toBytes might need to be tweaked to throw whatever ConfigException we throw now when bytes are out of range

Reply:
thank you, fixed